torch/distributed/utils.py

```diff
@@ -183,3 +183,67 @@ def _replace_by_prefix(
         new_key = new_prefix + key[len(old_prefix) :]
         state_dict[new_key] = state_dict[key]
         del state_dict[key]
+
+def apply_to_tensors(
+    fn: Callable,
+    container: Union[torch.Tensor, Dict, List, Tuple, Set, OrderedDict, PackedSequence],
+) -> Any:
+    """Recursively apply to all tensor in different kinds of container types."""
+
+    def apply(
```

Comment:
Btw `torch.to(obj, device)` then could be implemented in terms of this recursive `apply` function (https://github.com/pytorch/pytorch/issues/69431)